src/views/FlexExampleAppHomeView.mxml

```diff
@@ -1,204 +1,129 @@
-				trace("[INCENTIVIZED] DID_HIDE");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.DID_FAIL, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] DID_FAIL");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_STARTED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] ON_AUDIO_STARTED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_FINISHED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] ON_AUDIO_FINISHED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_AVAILABLE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_AVAILABLE");
-				if (IncentivizedAd.getInstance().isAvailable()) {
-					trace("Incentivized Ad is available.");
-				} else {
-					trace("Incentivized Ad is not available");
-				}
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.FETCH_FAILED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] FETCH_FAILED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_COMPLETED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_COMPLETED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_INCOMPLETE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_INCOMPLETE");
-				return null;
-			});
-			
-		]]>
+        import com.heyzap.sdk.ads.BannerAd;
+        import com.heyzap.sdk.ads.HeyzapAds;
+        import com.heyzap.sdk.ads.IncentivizedAd;
+        import com.heyzap.sdk.ads.InterstitialAd;
+        import com.heyzap.sdk.ads.VideoAd;
+
+        import model.AppModel;
+
+        [Bindable]
+        public var appModel:AppModel;
+
+        private var currentActionObject:*;
+
+        override public function set data(value:Object):void {
+            appModel = value as AppModel;
+        }
+
+        private function onCreationComplete():void {
+            currentActionObject = InterstitialAd.getInstance();
+        }
+
+        private function getEnteredAdTagOrNull():String{
+            if (!adTagTextInput.text) return null;
+            else return adTagTextInput.text;
+        }
+
+        private function onFetchButtonClick():void {
+            currentActionObject.fetch(getEnteredAdTagOrNull());
+        }
+
+        private function onHideButtonClick():void {
+            currentActionObject.hide();
+            appModel.bannerIsShown = false;
+        }
+
+        private function onShowButtonClick():void {
+            if (currentActionObject instanceof  BannerAd) {
+                currentActionObject.show(BannerAd.POSITION_BOTTOM,getEnteredAdTagOrNull()); //todo add position
+                appModel.bannerIsShown = true;
```

Comment:
but we use this only tho change show/hide button state. if banner fails user can just click hide and show again to try again. I think it's just test app, so it might be enough.
But if you want i can add additional listener for BannerAd to change buttons state on fail. Or just add some message to log - like "click hide and show again" if failed.please update